ambry-protocol/src/main/java/com.github.ambry.protocol/MessageInfoAndMetdataListSerde.java

```diff
@@ -146,12 +156,15 @@ void serializeMessageInfoAndMetadataList(ByteBuffer outputBuffer) {
         outputBuffer.putLong(messageInfo.getSize());
         outputBuffer.putLong(messageInfo.getExpirationTimeInMs());
         outputBuffer.put(messageInfo.isDeleted() ? UPDATED : (byte) ~UPDATED);
-        if (version < VERSION_1 || version > VERSION_5) {
+        if (version < VERSION_1 || version > VERSION_MAX) {
           throw new IllegalArgumentException("Unknown version in MessageInfoList " + version);
```

Comment:
Probably not related to your change. Just wonder why this check is not performed at the beginning of this method as  opposed to doing it in the whole loop.

Reply:
I didn't notice that, I suppose checking at the beginning of the method makes more sense. I will update.